frontend/src/core/user/components/FileUpload.js

```diff
@@ -0,0 +1,70 @@
+/* @flow */
+
+import * as React from 'react';
+import { Localized } from 'fluent-react';
+
+import './FileUpload.css';
+
+import type { NavigationParams } from 'core/navigation';
+
+
+type Props = {|
+    parameters: NavigationParams,
+|};
+
+type Ref = {
+    current: React.ElementRef<any> | null,
+};
+
+/*
+ * Render a File Upload button.
+ */
+export default class FileUpload extends React.Component<Props> {
+    uploadForm: Ref;
+
+    constructor(props: Props) {
+        super(props);
+        this.uploadForm = React.createRef();
+    }
+
+    submitForm = () => {
+        const current = this.uploadForm.current;
+        if (current) {
+            current.submit();
+        }
+    }
+
+    render() {
+        const { parameters } = this.props;
+
+        let csrfToken = '';
+        const rootElt = document.getElementById('root');
+        if (rootElt) {
+            csrfToken = rootElt.dataset.csrfToken;
+        }
+
+        return <form
+            action="/upload/"
+            className="uploadform"
+            encType="multipart/form-data"
+            method="POST"
+            ref={ this.uploadForm }
+        >
+            <input name="csrfmiddlewaretoken" type="hidden" value={ csrfToken } />
+            <input name="code" type="hidden" value={ parameters.locale } />
+            <input name="slug" type="hidden" value={ parameters.project } />
+            <input name="part" type="hidden" value={ parameters.resource } />
+            <label>
+                <Localized
+                    id="user-UserMenu--upload-translations"
+                    glyph={
+                        <i className="fa fa-cloud-upload-alt fa-fw"></i>
+                    }
+                >
+                    <span>{ '<glyph></glyph>Upload Translations' }</span>
```

Comment:
I believe it's needed. If I remove it, `<glyph></glyph>` doesn't render. Other `Localized` elements using `<glyph>` also wrap an element, not the text node directly. I also see the `<span>` in the DOM.

Reply:
Isn't that because the translation was missing in the `.ftl` file? I'm not very clear how fluent-react works internally but I don't understand why it would keep that span in the content. Discard my comment if you think it's not useful.

Reply:
Nope, it's in there already:
https://github.com/mozilla/pontoon/pull/1226/files#diff-141010cc4b360add776bdcffd473a577R160